lib/helpers.bash

```diff
@@ -155,60 +155,127 @@ _bash-it-plugins ()
     _bash-it-describe "plugins" "a" "plugin" "Plugin"
 }
 
-_bash-it_update() {
+_bash-it_update-dev() {
+  _about 'updates Bash-it to the latest master'
+  _group 'lib'
+
+  _bash-it_update- dev "$@"
+}
+
+_bash-it_update-stable() {
+  _about 'updates Bash-it to the latest tag'
+  _group 'lib'
+
+  _bash-it_update- stable "$@"
+}
+
+_bash-it_pull_and_update_inner() {
+  git checkout "$1" &> /dev/null
+  if [[ $? -eq 0 ]]; then
+    echo "Bash-it successfully updated."
+    echo ""
+    echo "Migrating your installation to the latest $2 version now..."
+    _bash-it-migrate
+    echo ""
+    echo "All done, enjoy!"
+    bash-it reload
+  else
+    echo "Error updating Bash-it, please, check if your Bash-it installation folder (${BASH_IT}) is clean."
+  fi
+}
+
+_bash-it_update-() {
   _about 'updates Bash-it'
+  _param '1: What kind of update to do (stable|dev)'
   _group 'lib'
 
+  declare silent
+  for word in $@; do
+    if [[ ${word} == "--silent" || ${word} == "-s" ]]; then
+      silent=true
+    fi
+  done
   local old_pwd="${PWD}"
 
   cd "${BASH_IT}" || return
 
-  if [ -z $BASH_IT_REMOTE ]; then
+  if [ -z "$BASH_IT_REMOTE" ]; then
     BASH_IT_REMOTE="origin"
   fi
 
-  git fetch &> /dev/null
+  git fetch $BASH_IT_REMOTE --tags &> /dev/null
 
+  if [ -z "$BASH_IT_DEVELOPMENT_BRANCH" ]; then
+    BASH_IT_DEVELOPMENT_BRANCH="master"
+  fi
+  # Defaults to stable update
+  if [ -z "$1" ] || [ "$1" == "stable" ]; then
+    version="stable"
+    TARGET=$(git describe --tags "$(git rev-list --tags --max-count=1)" 2> /dev/null)
+
+    if [[ -z "$TARGET" ]]; then
+      echo "Can not find tags, so can not update to latest stable version..."
+      return
+    fi
+  else
+    version="dev"
+    TARGET=${BASH_IT_REMOTE}/${BASH_IT_DEVELOPMENT_BRANCH}
+  fi
```

Comment:
This might be a little late to chime in, but what about offering the ability to choose any tag when not stable or dev? That way we cover momentary forced downgrade scenarios.

Reply:
Thanks for your feedback @cornfeedhobo !

I think this complicates this function too much. We assume that people usually do not want to downgrade, only in a case that the `dev` version is broken and they need a `stable` one. We should always make sure that the releases are stable. Your use case can be achieved by checking out the wanted git version and reloading, and fringe enough that people that want it can do it by themselves.

Reply:
Agreed, let's keep it simple!